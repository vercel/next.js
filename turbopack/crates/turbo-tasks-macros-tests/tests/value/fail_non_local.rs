#![feature(arbitrary_self_types)]
#![feature(arbitrary_self_types_pointers)]

use turbo_tasks::Vc;

#[turbo_tasks::value]
struct MyValue {
    value: Vc<i32>,
}

fn main() {
    let v = MyValue { value: Vc::cell(0) };
    let _ = v.value;
}
