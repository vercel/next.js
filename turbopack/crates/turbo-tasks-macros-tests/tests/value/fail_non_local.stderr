error[E0277]: the trait bound `Vc<i32>: NonLocalValue` is not satisfied
 --> tests/value/fail_non_local.rs:8:12
  |
8 |     value: Vc<i32>,
  |            ^^^^^^^ the trait `NonLocalValue` is not implemented for `Vc<i32>`
  |
  = help: the following other types implement trait `NonLocalValue`:
            &T
            &mut T
            ()
            (A, Z, Y, X, W, V, U, T)
            (B, A, Z, Y, X, W, V, U, T)
            (C, B, A, Z, Y, X, W, V, U, T)
            (D, C, B, A, Z, Y, X, W, V, U, T)
            (E, D, C, B, A, Z, Y, X, W, V, U, T)
          and $N others
note: required by a bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
 --> tests/value/fail_non_local.rs:6:1
  |
6 | #[turbo_tasks::value]
  | ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
  = note: this error originates in the derive macro `turbo_tasks::NonLocalValue` (in Nightly builds, run with -Z macro-backtrace for more info)
